text/0000-introduce empty ranges.md

```diff
@@ -0,0 +1,425 @@
+Another case in which a Range is now *empty* is when the bound equality `min == max` can be evaluated meaningfully, independent of whether `step` equals e.g. `0` or `+-Inf` such as for example `Range(10, 10, 0)`. In the case of equal and finite bounds, no value can exist that satisfies `[min, min)`. In case of infinite bound values, the range is also empty because one `inf` cannot be meaningfully tested for equality to another `inf` and even if it could, the second criterion would be violated in that a sequence of values would contain non-finite values.
+
+As already stated, the current `Range` implementation treats the occurrence of `+-Inf` or `NaN` parameters in *any* of the `min, max, step` parameters as sufficient condition for *infinite*. This RFC considers the opposite: the no-progress expression cannot be meaningfully evaluated if any one of the 3 tested parameters is a `NaN` value, and one can also not decide whether any .next() iterations which add NaN lie within the given numerical range, or, if `min` or `max` are `NaN`, what that range is in the first place. Any occurrence of `NaN` in the Range parameters therefore produces an *empty* Range. Similarly, while a `step` argument of `+-Inf` may "pass" the inspection by the *no-progress criterion*, no finite points within the mathematical range `[min, max)` can be computed, including when `min, max` themselves are `inf`. Therefore, such Ranges are *empty*, too. A `step` argument of `0` satisfies the *no-progress criterion* and produces an *empty* Range.
+
+The cases that would remain *infinite* under this proposal are then only those which fulfill the following condition: they are *not empty* (as per the evaluation above) AND their upper bound is either `+Inf` or `-Inf`. Typical examples would be `Range[F64](0, inf, 1)` or `Range[F64](0, -inf, -1)`. We can't allow a lower bound that is not finite, because then the elements returned on .next() calls would again not be finite.
+
+Summarizing this, the proposed algorithm for the Range classification is outlined in a close-to-code form. The following expressions are useful:
+
+```
+no_progress = ((min <= max) and (step <= 0) or (min >= max) and (step >= 0))
+progress = not no_progress = ((min > max) or (step > 0)) and ((min < max) or (step < 0))
+                           = (((min > max ) or (step > 0)) and (min < max)) or (((min > max ) or (step > 0)) and (step < 0))
+                           = ((step > 0) and (min < max)) or ((min > max ) and (step < 0))
+                           = ((min < max) and (step > 0)) or ((min > max ) and (step < 0))
```

Comment:
Why are you defining `progress` in terms of `not no_progress` only to provide additional cases where progress can be made? It seems redundant, and I misread it the first time as an instance of destructive read.

Reply:
I am not defining progress here -- it is calculated as `not no_progress`, because it is used in this positive form in the implementation